src/Runner.Common/HostContext.cs

```diff
@@ -641,6 +642,30 @@ public static HttpClientHandler CreateHttpClientHandler(this IHostContext contex
             var handlerFactory = context.GetService<IHttpClientHandlerFactory>();
             return handlerFactory.CreateClientHandler(context.WebProxy);
         }
+
+        public static string GetDefaultShellForScript(this IHostContext hostContext, string path, Common.Tracing trace, string prependPath)
```

Comment:
since you have the `hostContext` here, do we still need to pass in a `trace` object from the caller?

you can also create a new one via `var trace = hostContext.GetTrace(nameof(GetDefaultShellForScript));`

Reply:
Lets do this!